Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/cut_with_plane.h

```diff
@@ -0,0 +1,649 @@
+    for (halfedge_descriptor h : halfedges_around_target(hv, pm))
+    {
+      if (is_border(h, pm)) continue;
+      Oriented_side prev_ori = get(vertex_os, source(h, pm)),
+                    next_ori = get(vertex_os, target(next(h, pm), pm));
+      if ( prev_ori == ON_ORIENTED_BOUNDARY || next_ori == ON_ORIENTED_BOUNDARY) continue; // skip full edge
+      if (prev_ori!=next_ori) splitted_faces[face(h, pm)].push_back(h); // skip tangency point
+    }
+  }
+
+  //TODO: parallel for
+  for (std::pair<const face_descriptor, std::vector<halfedge_descriptor>>& f_and_hs : splitted_faces)
+  {
+    std::size_t nb_hedges = f_and_hs.second.size();
+
+    CGAL_assertion( nb_hedges%2 ==0 );
+
+    if (nb_hedges==2)
+    {
+      halfedge_descriptor h1=f_and_hs.second[0], h2=f_and_hs.second[1];
+      CGAL_assertion(next(h1,pm)!=h2 && next(h2,pm)!=h1); // the edge does not already exist
+      halfedge_descriptor res = CGAL::Euler::split_face(h1, h2, pm);
+      put(edge_is_marked, edge(res, pm), true);
+
+      if (triangulate)
+      {
+        if (!is_triangle(res, pm))
+        {
+          // TODO: take the criteria in triangulate_faces ?
+          halfedge_descriptor newh =
+            CGAL::Euler::split_face(res, next(next(res, pm), pm), pm);
+          put(edge_is_marked, edge(newh, pm), false);
+        }
+        else
+        {
+          res = opposite(res, pm);
+          if (!is_triangle(res, pm))
+          {
+            // TODO: take the criteria in triangulate_faces ?
+            halfedge_descriptor newh =
+              CGAL::Euler::split_face(res, next(next(res, pm), pm), pm);
+            put(edge_is_marked, edge(newh, pm), false);
+          }
+        }
+      }
+    }
+    else
+    {
+      // sort hedges to make them match
+      CGAL_assertion(!triangulate);
+      // TODO: need mechanism to make it robust even with EPICK
+      auto less_hedge = [&pm, vpm](halfedge_descriptor h1, halfedge_descriptor h2)
+      {
+        return lexicographically_xyz_smaller(get(vpm,target(h1,pm)), get(vpm,target(h2,pm)));
+      };
+      std::sort(f_and_hs.second.begin(), f_and_hs.second.end(), less_hedge);
+
+      for (std::size_t i=0; i<nb_hedges; i+=2)
+      {
+        halfedge_descriptor h1=f_and_hs.second[i], h2=f_and_hs.second[i+1];
+        CGAL_assertion(next(h1,pm)!=h2 && next(h2,pm)!=h1); // the edge does not already exist
+        halfedge_descriptor res = CGAL::Euler::split_face(h1, h2, pm);
+        put(edge_is_marked, edge(res, pm), true);
+      }
+    }
+  }
+}
+
+/**
+  * \ingroup PMP_corefinement_grp
+  *
+  * \brief clips `tm` by keeping the part that is on the negative side of `plane` (side opposite to its normal vector).
+  *
+  * If `tm` is closed, the clipped part can be closed too if the named parameter `clip_volume` is set to `true`.
+  * See Subsection \ref coref_clip for more details.
+  *
+  * \note `Plane_3` must be from the same %Kernel as the point of the internal vertex point map of `PolygonMesh`.
```

Comment:
Here as well I would suggest to lowercase the word "kernel"  as it is just a normal noun.